draft-ietf-quic-http.md

```diff
@@ -483,14 +483,22 @@ HTTP/3 uses QPACK header compression as described in [QPACK], a variation of
 HPACK which allows the flexibility to avoid header-compression-induced
 head-of-line blocking.  See that document for additional details.
 
-An HTTP/3 implementation MAY impose a limit on the maximum size of the header it
-will accept on an individual HTTP message; encountering a larger message header
-SHOULD be treated as a stream error of type `HTTP_EXCESSIVE_LOAD`.  If an
-implementation wishes to advise its peer of this limit, it can be conveyed as a
-number of bytes in the `SETTINGS_MAX_HEADER_LIST_SIZE` parameter. The size of a
-header list is calculated based on the uncompressed size of header fields,
-including the length of the name and value in bytes plus an overhead of 32 bytes
-for each header field.
+An HTTP/3 implementation MAY impose a limit on the maximum size of the message
+header it will accept on an individual HTTP message.  A server that receives a
+larger header field list than it is willing to handle can send an HTTP 431
+(Request Header Fields Too Large) status code {{?RFC6585}}.  A client can
+discard responses that it cannot process.  The size of a header field list is
+calculated based on the uncompressed size of header fields, including the length
+of the name and value in bytes plus an overhead of 32 bytes for each header
+field.
+
+If an implementation wishes to advise its peer of this limit, it can be conveyed
+as a number of bytes in the `SETTINGS_MAX_HEADER_LIST_SIZE` parameter. An
+implementation which has received this parameter SHOULD NOT send an HTTP message
+header which exceeds the indicated size, as this will likely produce an error
+and could disrupt the entire connection if the peer reacts negatively.  However,
```

Comment:
nit: reacts negatively is a bit vague.  How about "could disrupt other outstanding requests if the peer chooses to close the connection in response."

Reply:
Actually, I think we’d better delete “and could disrupt the entire connection if the peer reacts negatively” considering the fact that such behavior goes against the spirit of this setting value being an advisory one
.

Reply:
It's advisory, but I think the spirit is that eventually, this CID isn't going to route correctly.  Every deployment has this property, it's just a matter of timeframe.  

So it is advisory.  But in the same sense that GOAWAY indicates not to send new requests, this is saying "Please don't send any more packets with CIDs below this index, because it might not work."

Reply:
@ianswett This issue is about an _HTTP_ setting. Maybe you're comment is for one of the "Retire CID" issues of the transport?

Reply:
Yes, nevermind.

Reply:
In previous discussions, I thought we concluded closing the connection if the limit was exceeded was acceptable and therefore this wasn't really advisory?

Reply:
It is advisory, in that you're being told in advance what message size the peer will refuse to process.  When this was recommended to be a stream error, the risk was that a stream error can always be promoted to a connection error at implementation discretion, and a connection error would tear everything else down with it.

Now that we recommend an HTTP status code instead, this language might be unnecessary.